local-cluster/tests/local_cluster.rs

```diff
@@ -1536,20 +1538,42 @@ fn test_wait_for_max_stake() {
     solana_logger::setup_with_default(RUST_LOG_FILTER);
     let validator_config = ValidatorConfig::default_for_test();
     let slots_per_epoch = MINIMUM_SLOTS_PER_EPOCH;
+    // Set this large enough to allow for skipped slots but still be able to
+    // make a root and derive the new leader schedule in time.
+    let stakers_slot_offset = slots_per_epoch.saturating_mul(MAX_LEADER_SCHEDULE_EPOCH_OFFSET);
+    // Reduce this so that we can complete the test faster by advancing through
+    // slots/epochs faster. But don't make it too small because it can make us
+    // susceptible to skipped slots and the cluster getting stuck.
+    let ticks_per_slot = 16;
     let mut config = ClusterConfig {
         cluster_lamports: DEFAULT_CLUSTER_LAMPORTS,
         node_stakes: vec![DEFAULT_NODE_STAKE; 4],
         validator_configs: make_identical_validator_configs(&validator_config, 4),
         slots_per_epoch,
-        stakers_slot_offset: slots_per_epoch,
+        stakers_slot_offset,
+        ticks_per_slot,
         ..ClusterConfig::default()
     };
     let cluster = LocalCluster::new(&mut config, SocketAddrSpace::Unspecified);
     let client = RpcClient::new_socket(cluster.entry_point_info.rpc().unwrap());
 
-    assert!(client
-        .wait_for_max_stake(CommitmentConfig::default(), 33.0f32)
-        .is_ok());
+    // This is based on the percentage of stake that is allowed to be activated
+    // each epoch.
+    let num_expected_epochs = 14;
```

Comment:
If you wanna be fancy can derive this from the constant in sdk:
```
let num_expected_epochs = 3f64.log(1. + NEW_WARMUP_COOLDOWN_RATE).ceil() as u32 + 1;
```

Reply:
The 3 here is essentially the 3 non-bootstrap validators, right?

Does this compute the time for the stake to completely activate? Because I think the test just waits for the bootstrap node stake to fall below 1/3, which would be significantly sooner

Reply:
Also, I'm not sure if we need the +1 because this isn't trying to find the epoch in which we will hit the threshold. It's just computing number of epochs it will take once we start activating the stake

Reply:
ya the 3 is from the 1/3. Original equation is 
![image](https://github.com/user-attachments/assets/09c8b838-220e-4d8b-ab04-a18898fc032d)
Where:
- `S` is `DEFAULT_NODE_STAKE`
- `R` is 9% aka `NEW_WARMUP_COOLDOWN_RATE`
- `X` is `num_expected_epochs`

Essentially the denominator represents the total stake of the system at each epoch, it increases by 9%

Reply:
Do we allow 9% of the total stake to be activated, or 9% of the stake that wants to be activated per epoch?